File: Daily Test/Java/Maximum Value - Add Or Product 12.12.2025.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        for (int x : arr) {
            int sum = 0, prod = 1;
            int temp = x;

            while (temp > 0) {
                int d = temp % 10;
                sum += d;
                prod *= d;
                temp /= 10;
            }

            System.out.print(Math.max(sum, prod) + " ");
        }
    }
}

File: Daily Test/C++/Maximum Value - Add Or Product 12.12.2025.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int arr[n];
    for(int i = 0; i < n; i++)
        cin >> arr[i];

    for(int x : arr) {
        int sum = 0, prod = 1;
        int temp = x;

        while (temp > 0) {
            int d = temp % 10;
            sum += d;
            prod *= d;
            temp /= 10;
        }

        cout << max(sum, prod) << " ";
    }
    return 0;
}